src/ds/thread_messaging.h

```diff
@@ -285,8 +285,9 @@ namespace threading
       return tasks[tid];
     }
 
-    bool run_one(Task& task)
+    bool run_one()
```

Comment:
I believe it is neater if this function does not take any argument and the task automatically deduced from the current thread ID calling `run_one()`.

Reply:
So there is a reason for this, and this change introduces a performance hit on a very hot path

calling get_current_thread_id() is a is a map lookup, this changes running a task from O(1) to O(log(n)).

When this was done we had an issue with thread_local and OE. This should have been resolved after the work Eddy did a couple of months ago. So the correct fix is to create a thread local variable that is this threads task.

Reply:
Afaict, the only place we were calling this `run_one` was from `enclave.h`, where we were also calling exactly the same pattern of functions so incurring the same lookup cost? So this shouldn't be any slower, and looks like a safer API (you can't pass another thread's `Task`).

Reply:
This is followed-up in #1672